src/plugins/data_views/common/data_views/data_view_lazy.ts

```diff
@@ -0,0 +1,619 @@
+          customLabel: this.fieldAttrs?.[field.name]?.customLabel,
+          customDescription: this.fieldAttrs?.[field.name]?.customDescription,
+          shortDotsEnable: this.shortDotsEnable,
+        });
+        this.fieldCache.set(field.name, fld);
+      }
+      dataViewFields[field.name] = fld;
+    });
+
+    return dataViewFields;
+  }
+
+  getScriptedFieldsForQuery() {
+    return Object.values(this.scriptedFieldsMap).reduce<Record<string, estypes.ScriptField>>(
+      (scriptFields, field) => {
+        scriptFields[field.name] = {
+          script: {
+            source: field.script as string,
+            lang: field.lang,
+          },
+        };
+        return scriptFields;
+      },
+      {}
+    );
+  }
+
+  async getComputedFields({ fieldNames = ['*'] }: { fieldNames: string[] }) {
+    const scriptFields: Record<string, estypes.ScriptField> = {};
+
+    const fieldMap = (
+      await this.getFields({
+        fieldName: fieldNames,
+        fieldTypes: ['date', 'date_nanos'],
+        scripted: false,
+        runtime: false,
+      })
+    ).getFieldMap();
+
+    const docvalueFields = Object.values(fieldMap).map((dateField) => {
+      return {
+        field: dateField.name,
+        format:
+          dateField.esTypes && dateField.esTypes.indexOf('date_nanos') !== -1
+            ? 'strict_date_time'
+            : 'date_time',
+      };
+    });
+
+    each(this.getScriptedFields({ fieldName: fieldNames }), function (field) {
+      scriptFields[field.name] = {
+        script: {
+          source: field.script as string,
+          lang: field.lang,
+        },
+      };
+    });
+
+    return {
+      scriptFields,
+      docvalueFields,
+      runtimeFields: this.getRuntimeMappings(),
+    };
+  }
+
+  getRuntimeMappings(): estypes.MappingRuntimeFields {
+    // @ts-expect-error composite type is not yet supported by es client but it can be forced
+    return this.runtimeFieldMap;
+  }
+
+  /**
+   * Creates static representation of the data view.
+   * @param includeFields Whether or not to include the `fields` list as part of this spec. If not included, the list
+   * will be fetched from Elasticsearch when instantiating a new Data View with this spec.
+   */
+  public async toSpec(params?: { fieldParams?: GetFieldsParams }): Promise<DataViewSpec> {
+    const spec = this.toSpecShared(!!params?.fieldParams);
+
+    if (params?.fieldParams) {
+      const fields: DataViewFieldSpecMap = {};
+      const fieldMap = (await this.getFields(params.fieldParams)).getFieldMap();
+      Object.values(fieldMap).forEach((field) => {
+        fields[field.name] = field.toSpec({
+          getFormatterForField: this.getFormatterForField.bind(this),
+        });
+      });
+      spec.fields = fields;
+    }
+
+    return spec;
+  }
+
+  /**
+   * Creates a minimal static representation of the data view. Fields and popularity scores will be omitted.
+   */
+  public toMinimalSpec(): Omit<DataViewSpec, 'fields'> {
```

Comment:
I think we'll need to update this implementation of `toMinimalSpec` to include the changes from https://github.com/elastic/kibana/pull/178983.